### ISSUE TITLE
Add auc calculation operator ?

### ISSUE THREAD

**Issue by chenghuige** [tensorflow/tensorflow#397]
Any plans for adding this ?  Like in sklearn   roc_auc_score(teY, predicts)

**Comment by zhangchen-qinyinghua** [comment]
Is it that hard to add AUC calc and optimize by hand, I wonder?
In my opinion, making a continuous approximation of the AUC function will just do.

When I was using Theano, I tried to add the AUC loss with something related to some Mann Whitney Kernel, but I haven't tried that using TF yet.

**Comment by chenghuige** [comment]
@zhangchen-qinyinghua  Yes, seems for tensorflow it is very convenient to implement user_op of this kind, closing this issue.

**Comment by tobegit3hub** [comment]
`AUC` is commonly used in LR or DNN models and I don't think everyone need to implement the user op again.

Now TF learn has implemented the `streaming_auc` in `tensorflow/tensorflow/contrib/metrics/python/ops/metric_ops.py`. We will test and hope it could be used out of the scope of TF learn API.

**Comment by crystalmswang** [comment]
The auc for binary is implemented, but for multi-label isn't. I am doing a multi-label classification task and need to compute the acpr, but no method to slove it .
